### PR TITLE
Remove support for unstable private read receipts

### PR DESCRIPTION
**Pull request by SimonBrandner**
Fixes https://github.com/matrix-org/synapse/issues/13468

<hr>

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog).
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Review by erikjohnston — APPROVED** [review]
Woop, thanks!

File: synapse/handlers/receipts.py [hunk 1]
```diff
@@ -206,38 +203,24 @@ def filter_out_private_receipts(
             for event_id, orig_event_content in room.get("content", {}).items():
                 event_content = orig_event_content
                 # If there are private read receipts, additional logic is necessary.
-                if (
-                    ReceiptTypes.READ_PRIVATE in event_content
-                    or ReceiptTypes.UNSTABLE_READ_PRIVATE in event_content
-                ):
```
Comment (clokep) [inline synapse/handlers/receipts.py:212]: I think this means that we'll now leak any previous unstable private read receipts, which is a huge privacy leak.

I think we should nuke them from the database using a delta file.

Comment (clokep) [reply synapse/handlers/receipts.py:212]: See #13692.

Comment (SimonBrandner) [reply synapse/handlers/receipts.py:212]: Ah, sorry, I managed to forget about it again...

**Pull request merged**